lib/scenario_runner.py

```diff
@@ -1304,6 +1314,48 @@ def _setup_services(self):
+            # Docker run -d returns exit code 0 (success) even when containers fail moments later
+            # Common causes: architecture mismatch, missing dependencies, invalid entrypoints
+            time.sleep(1)
+            check_ps = subprocess.run(
+                    ['docker', 'ps', '-q', '-f', f'name={container_name}'],
+                    capture_output=True,
+                    text=True,
+                    check=False
+                )
+            if not check_ps.stdout.strip():
+                # Container not running anymore!
+                logs_ps = subprocess.run(
+                    ['docker', 'logs', container_name],
+                    capture_output=True,
+                    text=True,
+                    check=False
+                )
+                inspect_ps = subprocess.run(
+                    ['docker', 'inspect', '--format={{.State.ExitCode}}', container_name],
+                    capture_output=True,
+                    text=True,
+                    check=False
+                )
+                exit_code = inspect_ps.stdout.strip() if inspect_ps.returncode == 0 else "unknown"
+
+                if exit_code == "0":
+                    # Container exited successfully but immediately - unexpected behavior, log but continue
+                    error_helpers.log_error(f"Container '{container_name}' exited immediately after start with success code",
+                                          container_name=container_name,
+                                          exit_code=exit_code,
+                                          reason="Container completed and exited immediately (e.g., hello-world, simple commands) or due to configuration issues (invalid/missing entrypoint, incorrect command)",
+                                          container_logs_stdout=logs_ps.stdout,
+                                          container_logs_stderr=logs_ps.stderr)
+                else:
+                    # Container failed with non-zero or unknown exit code - this is a real failure
+                    image_arch, host_arch, is_compatible = self._validate_image_architecture(service['image'], raise_on_mismatch=False)
+                    if image_arch and host_arch and not is_compatible:
+                        raise RuntimeError(f"Container '{container_name}' failed immediately after start, probably due to architecture incompatibility (exit code: {exit_code}). Image architecture is '{image_arch}' but host architecture is '{host_arch}'.\nContainer logs:\n{logs_ps.stdout}\n{logs_ps.stderr}")
+                    else:
+                        raise RuntimeError(f"Container '{container_name}' failed immediately after start (exit code: {exit_code}). This indicates startup issues such as missing dependencies, invalid entrypoints, or configuration problems.\nContainer logs:\n{logs_ps.stdout}\n{logs_ps.stderr}")
+
             print('Stdout:', container_id)
```

Comment:
If this state is reached via emulation I would like to see a warning still. Please just add the arch incompatibility to the '_warnings' data which will later be saved via GMT